rasa_core/domain.py

```diff
@@ -182,27 +190,34 @@ def __init__(self,
                  slots,  # type: List[Slot]
                  templates,  # type: Dict[Text, Any]
                  action_names,  # type: List[Text]
+                 form_names,  # type: List[Text]
                  store_entities_as_slots=True,  # type: bool
                  restart_intent="restart"  # type: Text
                  ):
         # type: (...) -> None
 
         self.intent_properties = intent_properties
         self.entities = entities
+        self.form_names = form_names
         self.slots = slots
         self.templates = templates
 
         # only includes custom actions and utterance actions
         self.user_actions = action_names
-        # includes all actions (custom, utterance, and default actions)
+        # includes all actions (custom, utterance, default actions and forms)
         self.action_names = action.combine_user_with_default_actions(
-                action_names)
-
+                action_names) + self.form_names
```

Comment:
I think it is less confusing:
```suggestion
                action_names) + form_names
```

Reply:
what a feature